WordPress/Classes/ViewRelated/NUX/LoginSocialErrorViewController.swift

```diff
@@ -0,0 +1,153 @@
+
+        super.init(coder: aDecoder)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        tableView.estimatedRowHeight = 100.0
+
+        view.backgroundColor = WPStyleGuide.greyLighten30()
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard indexPath.section == Sections.buttons.rawValue else {
+            return
+        }
+
+        let controllerKey: String
+        switch indexPath.row {
+        case Buttons.tryEmail.rawValue:
+            controllerKey = "emailEntry"
+        case Buttons.tryAddress.rawValue:
+            controllerKey = "siteAddress"
+        case Buttons.signup.rawValue:
+            fallthrough
+        default:
+            controllerKey = "SignupViewController"
+        }
+
+        let storyboard = UIStoryboard(name: "Login", bundle: nil)
+        if let controller = storyboard.instantiateViewController(withIdentifier: controllerKey) as? NUXAbstractViewController {
+            navigationController?.setViewControllers([controller], animated: true)
+        }
+    }
+}
+
+
+// MARK: UITableViewDelegate methods
+
+extension LoginSocialErrorViewController {
+    override func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 45.0
+    }
+
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+}
+
+
+// MARK: UITableViewDataSource methods
+
+extension LoginSocialErrorViewController {
+    private struct Constants {
+        static let buttonCount = 3
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return Sections.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case Sections.titleAndDescription.rawValue:
+            return 1
+        case Sections.buttons.rawValue:
+            return Constants.buttonCount
+        default:
+            return 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        switch indexPath.section {
+        case Sections.titleAndDescription.rawValue:
+            cell = titleAndDescriptionCell()
+        case Sections.buttons.rawValue:
+            fallthrough
+        default:
+            cell = buttonCell(index: indexPath.row)
+        }
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        let footer = UIView()
+        footer.backgroundColor = WPStyleGuide.greyLighten20()
+        return footer
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        return 0.5
```

Comment:
I think this method can be removed if `tableView:viewForFooterInSection:` returns `UIView(frame: .zero)`. Seems that way when I test.  Same for setting the `footer.backgroundColor`

Reply:
This draws the dark line above and below the button cells.

Reply:
I totally missed that there were lines in the mock.  Totally makes sense now. :)

> The initial frame size of .zero will be ignored/modified by autolayout

This hasn't been my experience, but whatev 🤷‍♂️

Reply:
I usually just do `UIView()` not sure why I decided to use `UIView(frame: .zero)` here 😜 I'll probably switch it to the former, to make the result more clear.